### PR TITLE
Wasm-native stable memory dev blog post

### PR DESCRIPTION
**Pull request by adambratschikaye**
I'm not sure if changes are required to `CODEOWNERS` or `sidebar.js`.

### CODE REVIEW

File: blog/features/wasm-native-stable-memory.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+title: Wasm-native Stable Memory
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:2]:
```suggestion
title: Wasm-native stable memory
```

File: blog/features/wasm-native-stable-memory.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+description: "Introducing Wasm-native Stable Memory"
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:3]:
```suggestion
description: "Introducing Wasm-native stable memory"
```

File: blog/features/wasm-native-stable-memory.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+image: /img/blog/wasm-native-stable-memory-diagram-new.png
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:4]:
```suggestion
tags: [New features]
image: /img/blog/wasm-native-stable-memory-diagram-new.png
```

File: blog/features/wasm-native-stable-memory.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+---
+
+# Wasm-Native Stable Memory
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:7]:
```suggestion
# Wasm-native stable memory
```

File: blog/features/wasm-native-stable-memory.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+Wasm-native stable memory is a performance improvement to the Internet Computer’s stable memory interface which was recently rolled out to all subnets. It provides a 1.5-2x performance improvement to most stable memory workloads and does so in a completely transparent way. So if you’re already developing canisters that use stable memory then those canisters will automatically benefit from this performance improvement and if you’re not using stable memory, maybe now’s the time to try it out!
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:8]:
```suggestion
Wasm-native stable memory is a performance improvement to the Internet Computer’s stable memory interface, which was recently rolled out to all subnets. It provides a 1.5-2x performance improvement to most stable memory workloads and does so in a completely transparent way. 

If you’re already developing canisters that use stable memory, then those canisters will automatically benefit from this performance improvement. If you’re not using stable memory, maybe now’s the time to try it out!
```

File: blog/features/wasm-native-stable-memory.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+Each read from or write to stable memory goes through a system API call, and this call has a significant overhead. It first calls into the Wasmtime runtime, which then invokes code in the replica to copy over data from or to the stable memory. This indirection is needed because the running canister Wasm has no way to directly access the backing store of the stable memory.
+
+The following diagram shows the difference between accessing the main Wasm memory versus stable memory:
+
+![Original stable memory diagram](/img/blog/wasm-native-stable-memory-diagram-old.png)
+ 
+## How Wasm-native Stable Memory works
+
+Fortunately the new [multiple memories](https://github.com/WebAssembly/multi-memory/blob/master/proposals/multi-memory/Overview.md) Wasm feature allows a Wasm module to directly address multiple byte arrays (called “memories” in the Wasm spec). We can use this to modify calls to the stable memory APIs to directly read or write from a secondary Wasm memory.
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:38]:
```suggestion
Fortunately, the new [multiple memories](https://github.com/WebAssembly/multi-memory/blob/master/proposals/multi-memory/Overview.md) Wasm feature allows a Wasm module to directly address multiple byte arrays, called “memories” in the Wasm spec. We can use this module to modify calls to the stable memory APIs to directly read or write from a secondary Wasm memory.
```

File: blog/features/wasm-native-stable-memory.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+The following diagram shows how stable memory accesses will work under Wasm-native stable memory:
+
+![Wasm-native stable memory diagram](/img/blog/wasm-native-stable-memory-diagram-new.png)
+
+## Performance Gains on Sample App
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:44]:
```suggestion
## Performance gains on sample dapp
```

File: blog/features/wasm-native-stable-memory.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    MAP.with(|p| {
+        let mut map = p.borrow_mut();
+        for (k, v) in entries.into_iter() {
+            map.insert(k, v);
+        }
+    })
+}
+```
+
+After deploying this dapp to the IC, we then executed multiple messages that bulk insert many key-value pairs (close to maxing out the ingress message size limit). The following graphs show the execution time for each of these messages.
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:80]:
```suggestion
After deploying this dapp to the IC, we then execute multiple messages that bulk insert many key-value pairs (close to maxing out the ingress message size limit). The following graphs show the execution time for each of these messages.
```

File: blog/features/wasm-native-stable-memory.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+First without Wasm-native stable memory:
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:82]:
```suggestion
First, without Wasm-native stable memory:
```

File: blog/features/wasm-native-stable-memory.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+![Metrics without Wasm-native stable memory](/img/blog/wasm-native-stable-memory-execution-metrics-no-wnsm.png)
+
+Then with Wasm-native stable memory:
+
+![Metrics with Wasm-native stable memory](/img/blog/wasm-native-stable-memory-execution-metrics-wnsm.png)
+
+So we can see that this dapp gets a 1.5x speedup from Wasm-native stable memory.
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:90]:
```suggestion
We can see that this dapp gets a 1.5x speedup from using Wasm-native stable memory.
```

File: blog/features/wasm-native-stable-memory.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+## Production Examples
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:92]:
```suggestion
## Production examples
```

File: blog/features/wasm-native-stable-memory.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+If you’re interested in seeing some architecture of more complicated dapps that are using stable memory, the [bitcoin canister](https://github.com/dfinity/bitcoin-canister) and the [internet identity canister](https://github.com/dfinity/internet-identity/tree/main/src/internet_identity) are good examples.
```
Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:94]:
```suggestion
If you’re interested in seeing some architecture of more complicated dapps that are using stable memory, the [Bitcoin canister](https://github.com/dfinity/bitcoin-canister) and the [Internet Identity canister](https://github.com/dfinity/internet-identity/tree/main/src/internet_identity) are good examples.
```

Comment (jessiemongeon1) [inline blog/features/wasm-native-stable-memory.md:94]: Could you add a 'wrap up' or 'conclusion' section after this that closes out the blog post? As it stands, it feels like it ends a bit abruptly.

Comment (adambratschikaye) [reply blog/features/wasm-native-stable-memory.md]: I just added a short conclusion section.

**Pull request merged**